### ISSUE TITLE
sample-point works on threads? (Add to the README)

### ISSUE THREAD

**Issue by orena1** [BlueBrain/Fivox#91]
I am trying to use sample-point for many different locations, I built a python script which use subprocess.Popen to launch sample-point at different X Y Z (and make sure that only 16 are working simultaneously).

I am not sure, but I feel that running sample-point at a single node already tried to use all available threads at this node, am I right? If so it might be beneficial to add to the README that the tool is using all a threads (That it is parallel)

 Thanks.

**Comment by chevtche** [comment]
The loading of morphologies is multithreaded not the computation itself.

If you want to force it to use one thread only try:
OMP_NUM_THREADS=1

**Comment by orena1** [comment]
Thanks,
So it means that I should run 16 jobs per node (1 job per thread) to get the fastest results.

-----------
> if you want to force it to use one thread only try:
OMP_NUM_THREADS=1

This is very helpful because when I tried 16 jobs per node, It used all available PIDs and I was out of PID resources.

**Comment by chevtche** [comment]
Yes. But you won't win much since loading morphologies is the most costly operation in sample-points.
We should most probably implement an option to add multiple points in the command line. 

Maybe you can extract the curves you need from the volumes(voxelize) you generate with some python code. But you will need a lot of space if the time range is big.

**Comment by orena1** [comment]
> Maybe you can extract the curves you need from the volumes(voxelize) you generate with some python code. But you will need a lot of space if the time range is big.

I considered this option, but I did not managed to understand how to read the binary raw files with python, also I saw in the documentation the following:
```
The output is generated as a 3D volume, containing each of the voxels the LFP value corresponding to the position at the center of the voxel.

For completeness, a separate tool (sample-point) is provided, ..**. so the resulting values are more accurate, as an exact 3D position is used as input** ...
```

I did not manage to understand exactly why the results are more accurate in the sample-point? but I do want accurate results. 

I can overcome the storage problem, as I can do everything in batches, with python. I am trying to build a script that after each simulation will calculate the LFP in ~ 1000 points in the column and then to delete the current file.

Thanks.

**Comment by chevtche** [comment]
```
I considered this option, but I did not managed to understand how to read the binary raw files with python, also I saw in the documentation the following
```

The binary .raw files are just arrays of floats(by default). The .mhd should contain everything you need to compute the world coordinate from voxels coordinate. 
Do something like: origin + voxelCoord * spacing
I think there is also an .mhd loader in python. Maybe Taylor Newton has some code for doing that.

```
I did not manage to understand exactly why the results are more accurate in the sample-point? but I do want accurate results. 
```

It's not less accurate, it's just that the location of your "sample-point" are always located in the center of voxels in the volume, obviously. 
As an example in one 1d, with a volume, you will have locations: 0 1 2 3 4 5 if you use 6 voxels.
With sample-point you can put your probe anywhere like 1.3245 or 5.5. 
I don't think this is an issue for you.

I am leaving for holidays today. For 10 days. Someone can follow up our discussion  but it's not sure. We will think how to implement the multiple point command line parameter in the near future.

**Comment by eile** [comment]
Hi Oren: To extend a bit on what Greg was saying:

1) The volumes are written in the standardised mhd+raw format. The mhd file is human readable and contains the offset + spacing. There most certainly is an mhd reader in python.
2) To implement a ```sample-points``` tool we'll need to figure out a few details and we're not sure we've got the time before Greg is back. Any feedback on how urgent this is for you is appreciated.

**Comment by orena1** [comment]
Thanks Greg and Stefan,
This is not urgent to me, as you supplied me with a solution (I found a mhd reader), so whenever you like.

One last question, the raw file is a matrix, but the column itself is hexagon.
Is this means that some point which are sampled are outside of the column? 
Thanks for the support.

**Comment by eile** [comment]
Yes, we take the bounding box of the circuit to compute the volume. Naturally some points will be outside of the circuit.
